### PR TITLE
Trigger CiViCRM verification on successful oauth registration

### PR DESCRIPTION
**Pull request by verarojman**
Also reorganized Civicrm classes to ease the creation of a gem with those classes in the future.

Pending:

- [ ] Ensure user gets notification when automatically verified with CiViCRM

### CODE REVIEW

File: app/forms/decidim/verifications/civicrm.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require "digest/md5"
```
Comment (microstudi) [inline app/forms/decidim/verifications/civicrm.rb:3]:
```suggestion
require "digest"
```

File: config/locales/ca.yml [hunk 2]
```diff
@@ -25,5 +25,20 @@ ca:
             client_secret: Secret de client
             site: URL del servidor d'Omniauth
           enabled_by_default: Activat per defecte
+    verifications:
+      civicrm:
+        errors:
+          invalid_method: No us heu pogut verificar. El mètode de verificació OAuth2 no era "civicrm" tal com s'esperava.
+          invalid_status: No us heu pogut verificar. La verificació OAuth2 ha estat correcta.
```
Comment (microstudi) [inline config/locales/ca.yml:32]:
```suggestion
          invalid_status: No us heu pogut verificar. La verificació OAuth2 ha fallat.
```
?

File: config/locales/ca.yml [hunk 3]
```diff
@@ -25,5 +25,20 @@ ca:
...
+        invalid:
+          email_intro: No s'ha pogut concedir l'autorització de CiViCRM.
+          email_outro: Si us plau, poseu-vos en contacte amb el suport de la vostra plataforma per comprovar l'error que s'ha produït.
```
Comment (microstudi) [inline config/locales/ca.yml:35]:
```suggestion
          email_outro: Si us plau, poseu-vos en contacte amb el suport de la vostra plataforma per comprovar l'error que ha ocorregut.
```

File: config/locales/ca.yml [hunk 4]
```diff
@@ -25,5 +25,20 @@ ca:
...
+          email_subject: No us heu pogut verificar amb CiViCRM
```
Comment (microstudi) [inline config/locales/ca.yml:36]:
```suggestion
          email_subject: La verificació amb CiViCRM ha fallat
```

File: config/locales/ca.yml [hunk 5]
```diff
@@ -25,5 +25,20 @@ ca:
...
+          notification_title: No us heu pogut verificar amb CiViCRM
```
Comment (microstudi) [inline config/locales/ca.yml:37]:
```suggestion
          notification_title: La verificació amb CiViCRM ha fallat
```

File: config/locales/ca.yml [hunk 6]
```diff
@@ -25,5 +25,20 @@ ca:
...
+        ok:
+          email_intro: Se us ha concedit l'autorització amb CiViCRM.
```
Comment (microstudi) [inline config/locales/ca.yml:39]:
```suggestion
          email_intro: T'has autorizat correctament amb CiViCRM.
```

File: config/locales/ca.yml [hunk 7]
```diff
@@ -25,5 +25,20 @@ ca:
...
+          email_outro: Ara podeu realitzar totes les accions que requereixin l'autorització de CiViCRM.
+          email_subject: Has estat autoritzat amb CiViCRM.
+          notification_title: L'autenticació amb CiViCRM ha acabat amb èxit.
```
Comment (microstudi) [inline config/locales/ca.yml:42]:
```suggestion
          notification_title: Has sigut verificat amb CiViCRM correctament.
```

**Review by microstudi — COMMENTED** [review]
som suggestions for texts

File: app/forms/decidim/verifications/civicrm.rb (hunk 1 above)
Comment (verarojman) [reply app/forms/decidim/verifications/civicrm.rb:3]: Interesting, why is that better?

File: config/locales/ca.yml (hunk 3 above)
Comment (verarojman) [reply config/locales/ca.yml:39]: I think I would rather leave this one as it is.

File: config/initializers/decidim.rb [hunk 8]
```diff
@@ -128,10 +128,14 @@
...
 
   workflow.options do |options|
-    options.attribute :role, type: :enum, choices: -> { CivicrmApi::Models::User::ROLES.values.map(&:to_s) }
+    options.attribute :role, type: :enum, choices: -> { Decidim::Civicrm::Api::User::ROLES.values.map(&:to_s) }
   end
 end
+
+ActiveSupport::Notifications.subscribe(/^decidim\.user\.omniauth_registration/) do |_name, data|
+  Decidim::Civicrm::VerificationJob.perform_later(data[:user_id])
+end
```
Comment (microstudi) [inline config/initializers/decidim.rb:141]: We should move this to its own initializer file

**Pull request merged**